portfolio/src/main/webapp/script.js

```diff
@@ -37,9 +37,29 @@ async function showRandomFact() {
 
     // Pick a random fact
     const random_fact = facts[Math.floor(Math.random() * facts.length)];
-  
+
     // Display a random fact
     const factContainer = document.getElementById('fact-container');
     factContainer.innerText = random_fact;
-  }
-  
+}
+
+function requestTranslation() {
+    const text = document.getElementById('text').value;
+    const languageCode = document.getElementById('language').value;
+
+    const resultContainer = document.getElementById('result');
+    resultContainer.innerText = 'Loading...';
+
+    const params = new URLSearchParams();
+    params.append('text', text);
+    params.append('languageCode', languageCode);
+
+    fetch('/translate', {
+        method: 'POST',
+        body: params
+    }).then(response => response.text())
```

Comment:
Nit: the `.then` should have matching indentations. 
Example: 
```
someCall()
  .then(()=>{})
  .then(()=>{});
```
This prevents the indentation level from getting too crazy when chaining too many promises.

Reply:
Fixed!